### DISCUSSION TITLE
Compiling on Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by tkornack** [horizon-eda/horizon#774] [General]
I was attempting to compile on Ubuntu 24.04 and encountered an issue that the `liboce-ocaf-dev` package is no longer available. apt suggests that it is replaced by `libocct-ocaf-dev` and `libocct-data-exchange-dev`. It cheerfully compiles, but I'm getting linker errors on the XCAF* functions. Here's one of the error messages:

```
/usr/bin/ld: horizon-imp.p/src_import_step_step_importer.cpp.o: in function `horizon::STEPImporter::STEPImporter::get_shapes()':
/home/x/Repositories/horizon/build/../src/import_step/step_importer.cpp:465:(.text+0x1f68): undefined reference to `XCAFDoc_ShapeTool::GetFreeShapes(NCollection_Sequence<TDF_Label>&) const'
/usr/bin/ld: /home/x/Repositories/horizon/build/../src/import_step/step_importer.cpp:471:(.text+0x1fd4): undefined reference to `XCAFDoc_ShapeTool::GetShape(TDF_Label const&)'
```

**Comment by carrotIndustries** [comment]
Wroks in the CI: https://github.com/horizon-eda/horizon/actions/runs/8925848835/job/24515517038

**Comment by tkornack** [comment]
Indeed, it works. meson is mysterious to me; apparently if you install a dependency after the configuration stage it will never find it, even if you `meson setup build --reconfigure`. Removing the build directory entirely and recompiling worked.

It would be helpful to modify the documentation to reflect the packages that work on 24.04.
